system/ymodem/rb_main.c

```diff
@@ -176,40 +183,74 @@ int main(int argc, FAR char *argv[])
             strlcpy(ym_fd.pathname, optarg, PATH_MAX);
             if (ym_fd.pathname[strlen(ym_fd.pathname)] == '/')
               {
-                ym_fd.pathname[strlen(ym_fd.pathname)] = 0;
+                ym_fd.pathname[strlen(ym_fd.pathname)] = '\0';
               }
 
             break;
           case 'd':
-            ctx.recvfd = open(optarg, O_RDWR | O_NONBLOCK);
-            if (ctx.recvfd < 0)
+            recvfd = open(optarg, O_RDWR);
+            if (recvfd < 0)
               {
                 fprintf(stderr, "ERROR: can't open %s\n", optarg);
+                return recvfd;
               }
 
-            ctx.sendfd = ctx.recvfd;
             break;
-          case 'h':
-            show_usage(argv[0], EXIT_FAILURE);
+          case 'k':
+            customsize = atoi(optarg) * 1024;
             break;
-          default:
+          case 't':
+            timeout = atoi(optarg);
+            if (timeout != 0)
+              {
+                break;
+              }
+
+          case 'h':
           case '?':
-            fprintf(stderr, "ERROR: Unrecognized option\n");
+          default:
             show_usage(argv[0], EXIT_FAILURE);
             break;
         }
     }
 
-  ymodem_recv(&ctx);
-  if (ctx.recvfd)
+  ctx = malloc(sizeof(*ctx));
+  if (ctx == NULL)
     {
-      close(ctx.recvfd);
+      fprintf(stderr, "ERROR: can't malloc ymodem ctx\n");
+      ret = -ENOMEM;
+      goto out;
     }
 
+  memset(ctx, 0, sizeof(struct ymodem_ctx));
+  ctx->packet_handler = handler;
+  ctx->customsize = customsize;
+  ctx->timeout = timeout;
+  ctx->priv = &ym_fd;
+  if (recvfd)
+    {
+      ctx->recvfd = recvfd;
+      ctx->sendfd = recvfd;
```

Comment:
So both `recvfd`and `sendfd` are set to the same value? What is the point with that?